backend/webserver/views.py

```diff
@@ -67,3 +67,46 @@ class LogoutView(APIView):
     def post(self, request):
         logout(request)
         return Response({'message': 'Successfully Logged out'}, status=status.HTTP_200_OK)
+
+
+
+
+class PostView(APIView):
+    authentication_classes = [BasicAuthentication]
+    permission_classes = [permissions.IsAuthenticated]
+
+    def get_author(self,pk):
+        author = get_object_or_404(Author,pk=pk)
+        return author
+    def get_posts(self,post_id,author_id):
```

Comment:
You're getting a singular post here. So, method name should be `get_post` not `get_posts`